internal/neofs/tree.go

```diff
@@ -908,6 +916,47 @@ func (c *TreeClient) GetLock(ctx context.Context, cnrID *cid.ID, nodeID uint64)
 	return lockInfo, nil
 }
 
+func (c *TreeClient) GetObjectTaggingAndLock(ctx context.Context, cnrID *cid.ID, objVersion *data.NodeVersion) (map[string]string, *data.LockInfo, error) {
+	node, err := c.getTreeNodeWithKeys(ctx, cnrID, objVersion.ID, isTagKV, isLockKV)
```

Comment:
With latest changes I replaced `getTreeNodeWithKeys` with `getTreeNodes` and return map of two TreeNode structures and handle them separately. 

I will try to merge tags and locks into single TreeNode, but I am not sure if it worth it. Probably we can improve performance by having one tree node instead of two.